### PR TITLE
[release/6.0] HTTP/3: Use new QuicStream.ReadsCompleted property in transport

### LINKED ISSUE #35399: React to quic adding ReadsCompleted
React to https://github.com/dotnet/runtime/pull/57492 once we can ingest it

### PR DESCRIPTION
**Pull request by JamesNK**
Fixes #35399 Made this change in this branch first because main hasn't gotten latest runtime bits yet.

### CODE REVIEW

**Review by Tratcher — APPROVED** [review]
Does the Http/3 layer need any adjustments, or should that just work?

**Comment by JamesNK** [comment]
HTTP/3 layer will just work. It is already using the `ReadResult.IsCompleted` flag here:

https://github.com/dotnet/aspnetcore/blob/6350af9620165d6a1d65c7f3ce789a7909e95b1b/src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs#L428-L434

File: src/Servers/Kestrel/Transport.Quic/src/Internal/QuicStreamContext.cs [hunk 1]
```diff
@@ -240,12 +263,23 @@ private async Task DoReceive()
             finally
             {
                 // If Shutdown() has already bee called, assume that was the reason ProcessReceives() exited.
```
Comment (adityamandaleeka) [inline src/Servers/Kestrel/Transport.Quic/src/Internal/QuicStreamContext.cs:265]: not new code, but typo: "bee" -> "been"

**Review by adityamandaleeka — APPROVED** [review]

**Comment by JamesNK** [comment]
Closing this PR and have changes be merged from RC1 into back into 6.0 automatically.

**Pull request closed without merging**